modules/ensemble/lib/widget/button.dart

```diff
@@ -149,6 +155,24 @@ class ButtonState extends EWidgetState<Button> {
   }
 
   Widget _buildButtonChild() {
+    if (widget._controller.loading == true) {
+      return widget._controller.loadingWidget != null && scopeManager != null
+          ? scopeManager!
+              .buildWidgetFromDefinition(widget._controller.loadingWidget)
+          : // default loading widget
+          SizedBox(
```

Comment:
more of a design question for @amin-nas. Should the default loading state simply be the same button with reduced opacity? Probably the most common case that won't scream out "I need to update this right away".

Reply:
actually, the most common loading type is Circular loading, examples from our platform: MyPlace, UniConnect, and Chowmill all use circular loadings, and currently doing it with conditionals

![image](https://github.com/user-attachments/assets/6693d8f6-3a87-425f-bde8-7ef7be0e1e77)